clang-tools-extra/clangd/unittests/DiagnosticsTests.cpp

```diff
@@ -823,6 +823,20 @@ TEST(DiagnosticTest, ClangTidyNoLiteralDataInMacroToken) {
   EXPECT_THAT(TU.build().getDiagnostics(), UnorderedElementsAre()); // no-crash
 }
 
+TEST(DiagnosticTest, ClangTidyMacroToEnumCheck) {
+  Annotations Main(R"cpp(
+    #if 1
+    auto foo();
+    #endif
+  )cpp");
+  TestTU TU = TestTU::withCode(Main.code());
+  std::vector<TidyProvider> Providers;
+  Providers.push_back(addTidyChecks("cppcoreguidelines-macro-to-enum"));
```

Comment:
thanks for the test case, while at it, do you mind also adding `modernize-macro-to-enum` here :sweat_smile: